cve_bin_tool/version_compare.py

```diff
@@ -121,46 +83,44 @@ def version_compare(v1: str, v2: str):
                 if int(v1_array[i]) < int(v2_array[i]):
                     return -1
 
-            # If they're letters just do a string compare, I don't have a better idea
+            # If they're letters do a string compare.
             # This might be a bad choice in some cases: Do we want ag < z?
             # I suspect projects using letters in version names may not use ranges in nvd
             # for this reason (e.g. openssl)
             # Converting to lower() so that 3.14a == 3.14A
             # but this may not be ideal in all cases
             elif v1_array[i].isalpha() and v2_array[i].isalpha():
+                # allow pre-releases to come before arbitrary letters.
```

Comment:
A bit lost in this block here. I'm thinking of tests cases like:
- v1=pre4 & v2=3
- v1=2 & v2=pre3
Is the code doing the right thing for all of them (particularly the first one)?

Reply:
Dumped these in as explicit tests and they behave as I expect, so I guess I'm going to mark this as resolved.